text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,805 @@
+
+## Working Backwards
+
+### CHANGELOG
+
+`feat(cloudfront): s3 origin access control L2 construct`
+
+### README
+
+# Amazon CloudFront Construct Library
+
+Amazon CloudFront is a web service that speeds up distribution of your static and
+dynamic web content, such as .html, .css, .js, and image files, to your users.
+CloudFront delivers your content through a worldwide network of data centers called
+edge locations. When a user requests content that you're serving with CloudFront,
+the user is routed to the edge location that provides the lowest latency, so that
+content is delivered with the best possible performance.
+
+## Creating a Distribution
+
+CloudFront distributions deliver your content from one or more origins; an origin is
+the location where you store the original version of your content. Origins can be
+created from S3 buckets or a custom origin (HTTP server). Constructs
+to define origins are in the `aws-cdk-lib/aws-cloudfront-origins` module.
+
+Each distribution has a default behavior which applies to all requests to that
+distribution, and routes requests to a primary origin. Additional behaviors may
+be specified for an origin with a given URL path pattern. Behaviors allow routing
+with multiple origins, controlling which HTTP methods to support, whether to require
+users to use HTTPS, and what query strings or cookies to forward to your origin, among other settings.
+
+# CloudFront Origins for the CDK CloudFront Library
+
+## S3 Bucket
+
+An S3 bucket can be used as an origin. An S3 bucket origin can either be configured as a standard bucket or as a website endpoint (see [Use an S3 Bucket](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/DownloadDistS3AndCustomOrigins.html#using-s3-as-origin)).
+
+### Standard S3 Bucket
+
+To set up an origin using a standard S3 bucket, use the `S3BucketOriginWithOAI`, `S3BucketOriginWithOAC`, or `S3BucketOriginPublic` classes. The bucket
+is handled as a bucket origin and
+CloudFront's redirect and error handling will be used.
+
+### S3 Bucket Configured as a Website Endpoint
+
+To set up an origin using a S3 bucket configured as a website endpoint, use the `S3StaticWebsiteOrigin` class. When the bucket is configured as a
+website endpoint, the bucket is treated as an HTTP origin,
+and the distribution can use built-in S3 redirects and S3 custom error pages.
+
+```ts
+const myBucket = new s3.Bucket(this, 'myBucket');
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: new origins.S3StaticWebsiteOrigin(myBucket) },
+});
+```
+
+## Migrating from OAI to OAC
+
+If you are currently using OAI for your S3 origin and wish to migrate to OAC,
+replace the `S3Origin` construct (now deprecated) with `S3BucketOriginWithOAC`. You can create and pass in an `S3OriginAccessControl` or one will be automatically
+created by default.
+The OAI will be deleted as part of the
+stack update. The logical IDs of the resources managed by
```

Comment:
This is referring to retaining the CloudFront distribution and S3 bucket that they have set up already. Those will not get replaced as part of migrating from `S3Origin` with OAI to the new `S3BucketOrigin` setup

Reply:
mentioning the logical ID here is confusing because the origin and distribution are naturally separated. Can we instead say that this migration won't cause resource replacement?

Reply:
For sure, I'll reword it